### PR TITLE
First commit of plugin

### PR DESCRIPTION
**Pull request by jperedadnr**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,75 @@
...
+
+Run:
+
+    mvn clean client:compile
+
+The results will be available at `target/client/gvm`.
+
+#### `client:link`
+
+When the object is created, this goal will generate the native executable for the target platform.
```
Comment (abhinayagarwal) [inline README.md]: What is `object` referring to?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,75 @@
...
+        <configuration>
+            <mainClass>your.mainClass</mainClass>
+        </configuration>
+    </plugin>
+
+The plugin allows some options that can be set in `configuration`, to modify the default settings, and several goals, to build and run the native application.
+
+### 2. Goals
+
+You can run the regular goals to build and run your project as a regular Java project, if you use the `javafx-maven-plugin` plugin:
```
Comment (abhinayagarwal) [inline README.md]: May be re-write this as:

`javafx-maven-plugin` is internally applied by the `client-maven-plugin` allowing one to build and run the project using:

    mvn clean javafx:run

File: README.md [hunk 3]
```diff
@@ -1,2 +1,75 @@
...
+
+#### `client:link`
+
+When the object is created, this goal will generate the native executable for the target platform.
+
+Run:
+
+    mvn client:link
+    
+The results will be available at `target/client/$targetPlatform/$AppName.app`.
```
Comment (abhinayagarwal) [inline README.md]: `.app` is mac specific. May be we should just leave it as `$AppName`.

File: README.md (hunk 1 above)
Comment (jperedadnr) [reply README.md]: The `*.o` file that is generated after the AOT compilation phase. I'm not sure if we should specify it in more detail?

File: README.md (hunk 2 above)
Comment (jperedadnr) [reply README.md]: You actually have to combine both plugins. `client-maven-plugin` doesn't use `javafx-maven-plugin`, you still have to do `javafx:run` to run as usual on the VM, and `client:run` to run the native image.

File: README.md (hunk 3 above)
Comment (jperedadnr) [reply README.md]: ok

File: README.md (hunk 2 above)
Comment (abhinayagarwal) [reply README.md]: Ah, I see. My bad.

File: README.md (hunk 1 above)
Comment (abhinayagarwal) [reply README.md]: Can we just say the following?

> When the project is compiled for the target platform, this goal will generate the native executable

Comment (jperedadnr) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (jperedadnr) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (jperedadnr) [reply README.md]: I've tweaked it a little bit

File: LICENSE [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
```
Comment (abhinayagarwal) [inline LICENSE]: Not a very high priority, but can you remove the `/* */`?

Comment (jperedadnr) [reply LICENSE]: done

**Pull request merged**